GVFS/GVFS.Common/NuGetUpgrader/NuGetUpgrader.cs

```diff
@@ -0,0 +1,466 @@
+                        }
+                    }
+                }
+                catch (Exception ex)
+                {
+                    localError = ex.Message;
+                    installSuccessful = false;
+                }
+
+                if (!installSuccessful)
+                {
+                    activity.RelatedError($"Could not complete all install actions. The following error was encountered: {localError}");
+                    error = localError;
+                    return false;
+                }
+                else
+                {
+                    activity.RelatedInfo($"Install actions completed successfully.");
+                    error = null;
+                    return true;
+                }
+            }
+        }
+
+        public bool TrySetupToolsDirectory(out string upgraderToolPath, out string error)
+        {
+            return this.localUpgradeServices.TrySetupToolsDirectory(out upgraderToolPath, out error);
+        }
+
+        private static bool TryGetPersonalAccessToken(string gitBinaryPath, string credentialUrl, ITracer tracer, out string token, out string error)
+        {
+            GitProcess gitProcess = new GitProcess(gitBinaryPath, null, null);
+            return gitProcess.TryGetCredentials(tracer, credentialUrl, out string username, out token, out error);
+        }
+
+        private string UnzipPackageToTempLocation()
+        {
+            string extractedPackagePath = this.localUpgradeServices.TempPath;
+            ZipFile.ExtractToDirectory(this.DownloadedPackagePath, extractedPackagePath);
+            return extractedPackagePath;
+        }
+
+        public class NugetUpgraderConfig
+        {
+            private readonly ITracer tracer;
+            private readonly LocalGVFSConfig localConfig;
+
+            public NugetUpgraderConfig(ITracer tracer, LocalGVFSConfig localGVFSConfig)
+            {
+                this.tracer = tracer;
+                this.localConfig = localGVFSConfig;
+            }
+
+            public NugetUpgraderConfig(
+                ITracer tracer,
+                LocalGVFSConfig localGVFSConfig,
+                string feedUrl,
+                string packageFeedName,
+                string feedUrlForCredentials)
+                : this(tracer, localGVFSConfig)
+            {
+                this.FeedUrl = feedUrl;
+                this.PackageFeedName = packageFeedName;
+                this.FeedUrlForCredentials = feedUrlForCredentials;
+            }
+
+            public string FeedUrl { get; private set; }
+            public string PackageFeedName { get; private set; }
+            public string FeedUrlForCredentials { get; private set; }
+
+            public bool TryLoad(out bool isEnabled, out bool isCorrectlyConfigured, out string error)
+            {
+                error = string.Empty;
+
+                string configValue;
+                string readError;
+
+                bool feedURLAvailable = false;
```

Comment:
`Available` especially when using URL to me means that the URL it up and you can make requests and get responses from it but I don't think that's what it means in this context right?  It only means we could get a string from the config?  Maybe it should be `haveFeedUrl`?